### ISSUE TITLE
(unix) Test failures with Python 3.7

### ISSUE THREAD

**Issue by hroncok** [micropython/micropython#4000]
When I try to build MicroPython with Python 3.7 and I run the tests, I get:

```
cd ../../tests && MICROPY_MICROPYTHON=../ports/unix/micropython ./run-tests
Traceback (most recent call last):
  File "basics/gen_yield_from.py", line 20, in gen3
    raise StopIteration
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "basics/gen_yield_from.py", line 28, in <module>
    print(list(g))
  File "basics/gen_yield_from.py", line 24, in gen4
    print((yield from gen3()))
RuntimeError: generator raised StopIteration
Traceback (most recent call last):
  File "basics/gen_yield_from_close.py", line 62, in gen5
    yield 2
GeneratorExit

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "basics/gen_yield_from_close.py", line 65, in gen5
    raise StopIteration(123)
StopIteration: 123

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "basics/gen_yield_from_close.py", line 83, in <module>
    g.close()
  File "basics/gen_yield_from_close.py", line 72, in gen6
    print((yield from gen5()))
RuntimeError: generator raised StopIteration
Traceback (most recent call last):
  File "basics/gen_yield_from_throw.py", line 22, in gen3
    yield 123
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "basics/gen_yield_from_throw.py", line 28, in <module>
    g3.throw(StopIteration)
RuntimeError: generator raised StopIteration
Traceback (most recent call last):
  File "basics/generator_close.py", line 37, in gen2
    yield 1
GeneratorExit

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "basics/generator_close.py", line 40, in gen2
    raise StopIteration
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "basics/generator_close.py", line 44, in <module>
    print(g.close())
RuntimeError: generator raised StopIteration
...
12 tests failed: dict1 exception1 gen_yield_from gen_yield_from_close gen_yield_from_throw generator_close generator_return int_big_error set_pop subclass_native3 try_as_var sys_exc_info
```

When I set `MICROPY_CPYTHON3=python3.5` or `MICROPY_CPYTHON3=python3.6` the test pass.

This is related to [PEP 479](https://www.python.org/dev/peps/pep-0479/).

**Comment by dpgeorge** [comment]
Thanks for the report.  These errors are now "fixed" by 96e1fd480d6bc31b8c2954b20477cac262d1f1e5 through 828f771e327b932afc4865dbec53ce567dce45f5, which modify the tests and provide .py.exp where needed.

In addition to PEP 479, printing of exceptions with only one arg was changed, see https://bugs.python.org/issue30399

Since running the tests now works with Python 3.7 I'll close this issue.  Actually updating uPy to be compatible with Python 3.7 (eg implementing PEP479) would be separate issues.

**Comment by hroncok** [comment]
Thanks.
